book/platform/README.md

```diff
@@ -0,0 +1,473 @@
+
+<!-- $MDX file=examples/correct/hello-world/bin/main.ml -->
+```
+let () =
+  let greeting = Hello.greet "world" in
+  print_endline greeting
+```
+
+First perform a build of the project to generate the type annotation files. Then hover your mouse over the `Hello.greet` function -- you should see some documentation pop up about the function and its arguments.  This information comes from the _docstrings_ written into the `hello.mli` interface file in the library.
+
+<!-- $DISABLEDMDX file=examples/correct/hello-world/lib/hello.mli -->
+```
+(** This is a docstring, as it starts with "(**", as opposed to normal comments
+    that start with "(*".
+
+    The top-most docstring of the module should contain a description of the
+    module, what it does, how to use it, etc.
+
+    The function-specific documentation located below the function signatures. *)
+
+val greet : string -> string
+(** This is the docstring for the [greet] function.
+
+    A typical documentation for this function would be:
+
+    Returns a greeting message.
+
+    {4 Examples}
+
+    {[ print_endline @@ greet "Jane" ]} *)
+```
+
+Documentation strings are parsed by the [odoc](https://github.com/ocaml/odoc) tool generate HTML and PDF documentation from a collection of opam packages.  If you intend your code to be used by anyone else (or indeed, by yourself a few months later) you should take the time to annotate your OCaml signature files with documentation.  An easy way to preview the HTML documentation is to build it locally with dune:
+
+```skip
+$ opam install odoc
+$ dune build @doc
+```
+
+This will leave the HTML files in `_build/default/_doc/_html`, which you can view normally with a webbrowser.
+
+### Autoformatting your source code
+
+As you develop more OCaml code, you'll find it convenient to have it formatted to a common style.  The `ocamlformat` tool can help you do this easily from within VSCode.
+
+```skip
+$ echo 'version=0.19.0' > .ocamlformat
+$ opam install ocamlformat.0.19.0
+```
+
+The `.ocamlformat` file controls the autoformatting options available, and fixes the version of the tool that is used. You can upgrade to a newer ocamlformat version whenever you want, but it is a manual process to avoid an upstream release autoreformatting your project code without your intervention.  You can examine the formatting options via `ocamlformat --help` -- most of the time the defaults should be fine.
+
+Once you've got ocamlformat configured, you can either format your project from within VSCode (`shift-alt-F` being the default), or by running:
+
+```skip
+$ dune build @fmt
+```
+
+This will generate a set of reformatted files in the build directory, which you can accept with `dune promote` as you did earlier in the testing chapter.
```

Comment:
I'd maybe mention how to do this with one command: `dune build @fmt --auto-promote`. And that there is a command `fmt` in the Makefile of the project generated in the above section.